journal/202306.md

```diff
@@ -0,0 +1,861 @@
+- [Updated](https://github.com/decred/dcrd/commit/6c323c18b0eb7e4035c8ceb4ac3a36230c8c9c42) the `getwork` command and `notifywork` websocket notification. These are used by the mining code to fetch new data for hashing.
+- Added the [ASERT difficulty calculation](https://github.com/decred/dcrd/commit/d9b1921177d4626a22c380853d6c7030a2d5c854) and algorithm for miners to [generate blocks](https://github.com/decred/dcrd/commit/2db55eb6e7d16aca5676cd7d473555d23b30e23f) with the correct difficulty.
+- Implemented the DCP-11 [consensus vote](https://github.com/decred/dcrd/commit/85e3701a67fa5a9a7eb418ce71fa991ede447dbb). It will show up as `blake3pow` in voting software.
+
+In addition to the BLAKE3 PoW hash change which will make current Decred ASIC miners useless, the new ASERT algorithm will adjust the difficulty with each new block instead of every 144 blocks (the current adjustment frequency). For reference, Bitcoin's mining difficulty is updated every 2,016 blocks, or roughly every two weeks. ASERT stands for Absolutely Scheduled Exponentially weighted Rising Targets.
+
+The ASERT difficulty algorithm is also exponential, and can handle surging activity much better than the existing algorithm. This will greatly disincentivize "hit and run" miners who are only looking for a quick reward at a low difficulty. ASERT also better handles [large drops of hashrate](https://github.com/decred/dcrd/pull/3115#discussion_r1205741807) which cause slow blocks and degraded user experience. The network is stronger when miners are consistently creating new blocks and maintain the target block time of 5 minutes. See also the relevant [discussion on Matrix](https://matrix.to/#/!uCnoeYfbhUnvwWihzS:decred.org/$tbF9LyO1EzQ8m3JDHfX9EYnQz1zr8W0Ra73Yjtprj1M). For those who love fast math and bit tricks, make sure to read the [ASERT code here](https://github.com/decred/dcrd/blob/d9b1921177d4626a22c380853d6c7030a2d5c854/blockchain/standalone/pow.go#L222).
+
+{ _link to DCP-11 if available_ }
+
+{ _do we have the stats how much faster ASERT will adjust to a sharp spike/sharp drop of hashpower, compared to EMA?_ }
```

Comment:
The test data in the PR that implements it has various scenarios.  I'm not sure what number you're after here, but as noted elsewhere, the difficulty adjustment happens every block now instead of every 144, so, the difference can potentially be huge (exponentially so).

Reply:
Worst case for that example would be ~75 hours (~3.1 days) under the old algo versus 20-30 mins under the new algo.

Reply:
I tried looking at test data but could not figure out the practical difference. The improvement is mindblowing. I couldn't blink after reading this and my eyebrow muscles hurt now. So it will adjust in just a few of somewhat slow blocks. I [added](https://github.com/xaur/decred-news/commit/51bf8eb24c0bfba355624051f48891942f383248#diff-1d487d8831a0f2f7b2e5f0ea1543054a4e3b287dd31e36d115af3525a053b485R98) these numbers to the draft, this comparison is very enlightening.

Reply:
I think it's going to be super interesting to watch how the [graph of duration between blocks](https://dcrdata.decred.org/charts?chart=duration-btw-blocks&zoom=ikd7pc00-lk4o7400&bin=day&axis=time&visibility=true-false) smooths out over the coming months. Should be really cool.

Reply:
The overall distribution of the duration between blocks won't change much because mining still follows a Poisson distribution.

The overall expected values based on a perfect Poisson distribution are:

Elapsed | Expected
---------|----------
|10 sec|3.28%|
|30 sec|9.52%|
|1 min|18.13%|
|2 min|32.97%|
|3 min|45.12%|
|4 min|55.07%|
|5 min|63.21%|
|6 min|69.88%|
|7 min|75.34%|
|10 min|86.47%|
|20 min|98.17%|
|30 min|99.75%|

In other words, those are the percentages of blocks that should be found within the given time frames.